core/store/migrate/plugins/relayer/evm/migrate.go

```diff
@@ -0,0 +1,174 @@
+	// maybe update the cache to store the temp dir and delete it when cache is deleted
+	tmpDir, err := os.MkdirTemp("", cfg.Schema)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp dir: %w", err)
+	}
+
+	d := filepath.Join(tmpDir, cfg.Schema, cfg.ChainID.String())
+	err = os.MkdirAll(d, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create migration directory %s: %w", d, err)
+	}
+	migrations, err := generateMigrations(embeddedTmplFS, MigrationRootDir, d, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate migrations for opt %v: %w", cfg, err)
+	}
+	fmt.Printf("Generated migrations: %v\n", migrations)
+	fsys := os.DirFS(d)
+
+	// hack to avoid global state. the goose lib doesn't allow to disable the global registry
+	// and also pass custom go migrations (wtf the point of disabling the global registry then?)
+	// https://github.com/pressly/goose/blob/3ad137847a4c242f09e425a12c15c7c7961d4b0f/provider.go#L119
+	goose.ResetGlobalMigrations()
+	p, err := goose.NewProvider(
+		"",
+		db.DB, fsys,
+		goose.WithStore(store),
+		goose.WithDisableGlobalRegistry(true), // until/if we refactor the core migrations to use goose provider
+		goose.WithGoMigrations(goMigrations...))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create goose provider: %w", err)
+	}
+	providerCache[id] = p
+	return p, nil
+}
+
+/*
+func setupPluginMigrations(cfg Cfg) error {
+	// reset the base fs and the global migrations
+	goose.SetBaseFS(nil) // we don't want to use the base fs for plugin migrations because the embedded fs contains templates, not sql files
+	goose.ResetGlobalMigrations()
+	goose.SetTableName(fmt.Sprintf("goose_migration_relayer_%s_%s", cfg.Schema, cfg.ChainID.String()))
+	err := Register0002(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to register migration 0002: %w", err)
+	}
+	return nil
+}
+*/
+// Migrate migrates a subsystem of the chainlink database.
+// It generates migrations based on the template for the subsystem and applies them to the database.
+func Migrate(ctx context.Context, db *sqlx.DB, cfg Cfg) error {
+	p, err := newProvider(db, cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create goose provider: %w", err)
+	}
+	if todo, _ := p.HasPending(ctx); !todo {
+		return nil
+	}
+	/*
+		tmpDir, err := os.MkdirTemp("", cfg.Schema)
+		if err != nil {
+			return fmt.Errorf("failed to create temp dir: %w", err)
+		}
+		defer os.RemoveAll(tmpDir)
+
+		d := filepath.Join(tmpDir, cfg.Schema, cfg.ChainID.String())
+		err = os.MkdirAll(d, os.ModePerm)
+		if err != nil {
+			return fmt.Errorf("failed to create directory %s: %w", d, err)
+		}
+		migrations, err := generateMigrations(embeddedTmplFS, MigrationRootDir, d, cfg)
+		if err != nil {
+			return fmt.Errorf("failed to generate migrations for opt %v: %w", cfg, err)
+		}
+		fmt.Printf("Generated migrations: %v\n", migrations)
+	*/
```

Comment:
Getting a little confused about all of the commented code.  I see `setupPluginMigraitons is commented out, as well as `Register0002`, then this portion of `Migrate` which calls `generateMigrations` is commented, even though `generateMigrations` is uncommented? Which of these will we end up using, or are they all still WIP?